source/adapters/cuda/device.cpp

```diff
@@ -1009,7 +1009,7 @@ UR_APIEXPORT ur_result_t UR_APICALL urDeviceGetInfo(ur_device_handle_t hDevice,
           hDevice->get()));
     }
 
-    uint32_t MemoryBandwidth = MemoryClockKHz * MemoryBusWidth * 250;
+    uint64_t MemoryBandwidth = (uint64_t)MemoryClockKHz * MemoryBusWidth * 250;
```

Comment:
I seem to remember that there was an effort to get rid of c-style casts, so maybe:
```suggestion
    uint64_t MemoryBandwidth = static_cast<uint64_t>(MemoryClockKHz) * MemoryBusWidth * 250;
```
instead.

Reply:
The `static_cast` does the calculation before casting so it still overflows, unlike the c-style cast like here: https://godbolt.org/z/hdsP4K419 . I think I will define one of the variables as `uint64_t` and that should be a simpler option for that.

Reply:
That's only because you casted the result of calculation, not a single variable, see: https://godbolt.org/z/P8ocxPsar

Reply:
Ah okay, make sense. 

I changed the magic number `250` to a variable on its own, felt that would be a clearer solution. I tried to see what `250` mean, but it seems a mix of multiple constants multiplied together.

Reply:
I think you still need to flip the order of multiplication, to make sure that the 64-bit wide value is done first. The implicit conversions are done in order of evaluation of the expressions, so you need to make sure that all of the operations are done on widened, 64-bit, types. As it stands, it could still overflow.

https://godbolt.org/z/qn6K4Eh37

Reply:
Oh, Good shout, changed that.